komposable-architecture/src/main/java/com/toggl/komposable/architecture/Effect.kt

```diff
@@ -14,5 +21,25 @@ fun interface Effect<out Action> {
      * @return An action that will be sent again for further processing
      * @see Store.send
      */
-    suspend fun execute(): Action?
```

Comment:
should we also keep this to ease the migration?
```
interface Effect<out Action> {

    @Deprecated(
        "use invoke and return a [Flow] instead",
        replaceWith = ReplaceWith("invoke()")
    )
    suspend fun execute(): Action? = null

    /**
     * Executes the effect. This operation can produce side effects, and it's the
     * responsibility of the class implementing this interface to change threads
     * to prevent blocking the UI when needed.
     * @return An action that will be sent again for further processing
     * @see Store.send
     */
    fun invoke(): Flow<Action> = flow {
        @Suppress("DEPRECATION")
        execute()?.let { emit(it) }
    }
```

Reply:
This would be cool but we'd lose the `fun` modifier from the interface and break many things. I think we shouldn't look back and just provide `SuspendEffect` as a partial remedy instead of compromising effect's design going forward.